src/main/groovy/com/netflix/nebula/lint/rule/dependency/DependencyService.groovy

```diff
@@ -22,17 +22,23 @@ import java.util.jar.JarFile
 import java.util.zip.ZipException
 
 class DependencyService {
-    static Map<Project, DependencyService> instancesByProject = [:]
-    private final Logger logger = LoggerFactory.getLogger(DependencyService)
+    private static final String EXTENSION_NAME = "gradleLintDependencyService"
 
     static synchronized DependencyService forProject(Project project) {
-        def inst = instancesByProject[project]
-        if(inst) return inst
-        inst = new DependencyService(project)
-        instancesByProject[project] = inst
-        return inst
+        def extension = project.extensions.findByType(DependencyServiceExtension)
```

Comment:
Material code is just the top of the file, I formatted the file while I was at it.